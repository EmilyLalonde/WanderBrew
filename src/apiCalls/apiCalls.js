export const getBreweriesByState = async (state) => {
  const stateBreweries = `https://api.openbrewerydb.org/breweries?by_state=${state}&per_page=50`
  const response = await fetch(stateBreweries)
  if(!response.ok) {
    throw new Error('There was an error getting your albums')
  }
  const allStateBreweries = await response.json();
  return allStateBreweries
}

// export const getBreweriesByCity = async (city) => {
//   const cityBreweries = `https://api.openbrewerydb.org/breweries?by_name=${city}&per_page=50`
//   const response = await fetch(cityBreweries)
//   if(!response.ok) {
//     throw new Error('There was an error getting your albums')
//   }
//   const allCityBreweries = await response.json();
//   return allCityBreweries
// }


