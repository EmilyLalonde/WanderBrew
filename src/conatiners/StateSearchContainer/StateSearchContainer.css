.state-input-container {
  border: 2px solid black;
  margin-left: 1%;
  margin-top: 5%;
  height: 46px;
  width: 45%
}

.search {
  height: 40px;
  width: 83.9%
}

.search-button {
  background-color: #b8b8b8;
  border-left: 2px solid black;
  color: white;
  font-size: 22px;
  font-weight: 600;
  height: 46px;
  float: right;
  width: 15%;
}

.map-search-button:hover,
.search-button:hover {
  box-shadow:
  0 0 0.1vw  0.1vw #fff7f7,   
  0 0 0.3vw  0.3vw #002236, 
  0 0   1.5vw  0.2vw #b8b8b8,
  inset 0 0 1vw  0.2vw #b8b8b8,
  inset 0 0 0.3vw  0.1vw #002236,
  inset 0 0 0.1vw  0.1vw #fff7f7;
  background-color: rgba(0, 0, 0, .3)
}

.map-search-button.disabled,
.search-button.disabled {
  opacity: 0.65; 
  cursor: not-allowed;
}

h4 {
  font-size: 30px;
  height: 50px;
  margin-top: 10%;
  text-align: center;
}