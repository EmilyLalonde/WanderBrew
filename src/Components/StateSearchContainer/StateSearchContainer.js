import React, { Component } from 'react'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getStateResults } from '../../actions'
import './StateSearchContainer.css'
import { getBreweriesByState } from "../../apiCalls/apiCalls";
import StateContainer from '../StateContainer/StateContainer';

class StateSearchContainer extends Component {
  constructor() {
    super()
    this.state = {
      state: '',
      error: ''
    }
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmitState = async () => {
    this.setState({ state: "" });
    const { state } = this.state;
    const { getStateResults } = this.props
    getBreweriesByState(state)
    .then(data => getStateResults(data))
  }

  render() {
    return (
        <div>
        <input
          type="text"
          placeholder="Search for a State..."
          name="state"
          className="search"
          value={this.state.state}
          onChange={this.handleChange}
        />
        <button className="search-button" onClick={this.handleSubmitState}>
          Submit
        </button>
        <StateContainer />
      </div>
    )
  }
}

const mapStateToProps = state => ({
  stateResults: state.stateResults
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({ getStateResults }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(StateSearchContainer)