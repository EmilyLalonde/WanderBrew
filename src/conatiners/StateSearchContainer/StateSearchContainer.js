import React, { Component } from 'react'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getStateResults } from '../../actions'
import './StateSearchContainer.css'
import { getBreweriesByState } from "../../apiCalls/apiCalls";
import StateContainer from '../StateContainer/StateContainer';

class StateSearchContainer extends Component {
  constructor() {
    super()
    this.state = {
      state: '',
      error: '', 
      favorites: []
    }
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmitState = async () => {
    this.setState({ state: "" });
    const { state } = this.state;
    const { getStateResults } = this.props
    getBreweriesByState(state)
    .then(data => getStateResults(data))
  }

  render() {
    return (
      <main className="state-main">
        <div className="state-input-container">
        <input
          type="text"
          placeholder="Search for a State..."
          name="state"
          className="search"
          value={this.state.state}
          onChange={this.handleChange}
        />
        <button className="search-button" onClick={this.handleSubmitState}>
          Submit
        </button>
        </div>
        <div className="state-search-container">
        <StateContainer />
        </div>
        </main>
    )
  }
}

const mapStateToProps = state => ({
  stateResults: state.stateResults
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({ getStateResults }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(StateSearchContainer)