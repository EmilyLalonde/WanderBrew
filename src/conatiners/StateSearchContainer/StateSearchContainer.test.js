import React from 'react';
import { shallow } from 'enzyme';
import {StateSearchContainer, mapStateToProps } from './StateSearchContainer';
import { getBreweriesByName } from '../../apiCalls/apiCalls'

jest.mock('../../apiCalls/apiCalls');

getBreweriesByName.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

let mockGetBreweriesByName = jest.fn()

describe('StateSearchContainer', () => {
  let wrapper;
  let mockStateResults;

  beforeEach(() => {
    mockStateResults = [
      {
        id: 1516, 
        name: 'Ratio Beerworks', 
        street: '2920 Larimer St', 
        city: 'Denver', 
        state:'Colorado', 
        phone:'3012669264'
      }
    ]
    wrapper = shallow(<StateSearchContainer  getBreweriesByName={mockGetBreweriesByName}/>)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when handleChange is called', () => {
    wrapper = shallow(<StateSearchContainer  />)
    const mockEvent = { target: { name: 'state', value: 'Colorado'} };
    const expected = 'Colorado';

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('state')).toEqual(expected);
  });

  it('should run handleChange when the inputs detect a change', () => {
    const mockEvent = { target: { name: 'state', value: 'Colorado'} };
  
    wrapper.instance().handleChange = jest.fn();
    wrapper.find('[name="state"]').simulate('change', mockEvent);
  
    expect(wrapper.instance().handleChange).toHaveBeenCalledTimes(0);
  });

  it('should run handleSubmitState when the button is clicked', () => {
    const mockEvent =  jest.fn();

    wrapper.instance().handleSubmitState = jest.fn();
    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().handleSubmitState).toHaveBeenCalledTimes(0);
  });
})

