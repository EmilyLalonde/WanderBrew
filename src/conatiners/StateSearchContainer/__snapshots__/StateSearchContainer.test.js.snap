// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateSearchContainer should match the snapshot 1`] = `
<main
  className="state-main"
>
  <div
    className="state-input-container"
  >
    <input
      className="search"
      name="state"
      onChange={[Function]}
      placeholder="Search for a State..."
      type="text"
      value=""
    />
    <button
      className="search-button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="state-search-container"
  >
    <Connect(StateContainer) />
    <h4>
      ^ Search for Breweries by State ^
    </h4>
  </div>
</main>
`;
